Gemfile

```diff
@@ -35,62 +38,70 @@ end
+  gem 'beaker', *location_for(ENV['BEAKER_VERSION'] || '< 3')                    if Gem::Version.new(RUBY_VERSION) < Gem::Version.new('2.3.0') and ! supports_windows
+  gem 'beaker-pe',                                                               :require => false if Gem::Version.new(RUBY_VERSION) >= Gem::Version.new('2.3.0')
+  gem 'beaker-rspec', *location_for(ENV['BEAKER_RSPEC_VERSION'] || '>= 3.4')     if ! supports_windows
+  gem 'beaker-rspec', *location_for(ENV['BEAKER_RSPEC_VERSION'] || '~> 5.1')     if supports_windows
+  gem 'beaker-puppet_install_helper',                                            :require => false
+  gem 'master_manipulator',                                                      :require => false
+  gem 'beaker-hostgenerator', *location_for(ENV['BEAKER_HOSTGENERATOR_VERSION'])
 end
 
-# The recommendation is for PROJECT_GEM_VERSION, although there are older ways
-# of referencing these. Add them all for compatibility reasons. We'll remove
-# later when no issues are known. We'll prefer them in the right order.
-puppetversion = ENV['PUPPET_GEM_VERSION'] || ENV['GEM_PUPPET_VERSION'] || ENV['PUPPET_LOCATION'] || '>= 0'
-gem 'puppet', *location_for(puppetversion)
-
-# json_pure 2.0.2 added a requirement on ruby >= 2. We pin to json_pure 2.0.1
-# if using ruby 1.x
-gem 'json_pure', '<=2.0.1', :require => false if RUBY_VERSION =~ /^1\./
+gem 'puppet', *location_for(ENV['PUPPET_GEM_VERSION'])
 
 # Only explicitly specify Facter/Hiera if a version has been specified.
 # Otherwise it can lead to strange bundler behavior. If you are seeing weird
 # gem resolution behavior, try setting `DEBUG_RESOLVER` environment variable
 # to `1` and then run bundle install.
-facterversion = ENV['FACTER_GEM_VERSION'] || ENV['GEM_FACTER_VERSION'] || ENV['FACTER_LOCATION']
-gem "facter", *location_for(facterversion) if facterversion
-hieraversion = ENV['HIERA_GEM_VERSION'] || ENV['GEM_HIERA_VERSION'] || ENV['HIERA_LOCATION']
-gem "hiera", *location_for(hieraversion) if hieraversion
+gem 'facter', *location_for(ENV['FACTER_GEM_VERSION']) if ENV['FACTER_GEM_VERSION']
+gem 'hiera', *location_for(ENV['HIERA_GEM_VERSION']) if ENV['HIERA_GEM_VERSION']
+
 
 # For Windows dependencies, these could be required based on the version of
 # Puppet you are requiring. Anything greater than v3.5.0 is going to have
 # Windows-specific dependencies dictated by the gem itself. The other scenario
 # is when you are faking out Puppet to use a local file path / git path.
 explicitly_require_windows_gems = false
-puppet_gem_location = gem_type(puppetversion)
+puppet_gem_location = gem_type(ENV['PUPPET_GEM_VERSION'])
 # This is not a perfect answer to the version check
-if puppet_gem_location != :gem || puppetversion < '3.5.0'
+if puppet_gem_location != :gem or (Gem::Version.correct?(ENV['PUPPET_GEM_VERSION']) and Gem::Requirement.new('< 3.5.0').satisfied_by?(Gem::Version.new(ENV['PUPPET_GEM_VERSION'])))
```

Comment:
and then I noticed the control flow statements. https://github.com/puppetlabs/modulesync_configs/pull/99#discussion_r87042431